### PR TITLE
fix: fix timeout when append entries

### PR DESCRIPTION
**Pull request by iGxnon**
Signed-off-by: iGxnon <igxnon@gmail.com>

Please briefly answer these questions:

* what problem are you trying to solve? (or if there's no problem, what's the motivation for this change?)

Fix the lock order

* what changes does this pull request make?

* are there any non-obvious implications of these changes? (does it break compatibility with previous versions, etc)

No

### CODE REVIEW

File: crates/curp/src/server/raw_curp/mod.rs [hunk 1]
```diff
@@ -101,6 +101,8 @@ pub struct RawCurp<C: Command, RC: RoleChange> {
     cst: Mutex<CandidateState<C>>,
     /// Curp logs
     log: RwLock<Log<C>>,
+    /// Contexts of fallback log entries
+    fallback_contexts: DashMap<LogIndex, FallbackContext<C>>,
```
Comment (bsbds) [inline crates/curp/src/server/raw_curp/mod.rs:105]: Please use a `RwLock<HashMap>` for this, as we don't need to concurrently modify the map.

File: crates/curp/src/server/raw_curp/mod.rs [hunk 2]
```diff
@@ -747,6 +755,15 @@ impl<C: Command, RC: RoleChange> RawCurp<C, RC> {
             let mut log_w = RwLockUpgradableReadGuard::upgrade(log_r);
             if last_sent_index > log_w.commit_index {
                 log_w.commit_to(last_sent_index);
+                self.fallback_contexts.retain(|&idx, c| {
```
Comment (bsbds) [inline crates/curp/src/server/raw_curp/mod.rs:758]: Please move this to a seperate function.

**Comment by bsbds** [comment]
Please add more details about why the lock order could cause a timeout.

**Comment by Phoenix500526** [comment]
Close this PR since it doesn't provide enough information in the description. Additionally, there are no cases to demonstrate that this PR solves any problems.

**Pull request closed without merging**